great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -1473,19 +1473,20 @@ def _initialize_usage_statistics(
 
     def _init_datasources(self) -> None:
         """Initialize the datasources in store"""
-        datasource_keys: List[
-            Union[str, Tuple[str, str]]
-        ] = self._datasource_store.list_keys()
-
-        # In the case that our Store/StoreBackend generates tuple keys, we want to extract the datasource name
-        if datasource_keys and isinstance(datasource_keys[0], tuple):
-            datasource_keys = [key[1] for key in datasource_keys]
+        config: DataContextConfig = self.get_config_with_variables_substituted(
+            self.config
+        )
+        datasources: Dict[str, DatasourceConfig] = cast(
+            Dict[str, DatasourceConfig], config.datasources
+        )
 
-        for datasource_name in datasource_keys:
+        for datasource_name, datasource_config in datasources.items():
             try:
-                datasource: Optional[
-                    Union[LegacyDatasource, BaseDatasource]
-                ] = self.get_datasource(datasource_name=datasource_name)
+                config = copy.deepcopy(datasource_config)
+                config_dict = dict(datasourceConfigSchema.dump(config))
+                datasource = self._instantiate_datasource_from_config(
+                    name=datasource_name, config=config_dict
+                )
```

Comment:
This is all legacy code - instead of iterating through the store's keys (which makes `GET` requests in a Cloud-context), we simply want to use the config we have. There's no point in making subsequent requests if the first request we made as part of `DataContext.__init__` already has the information we need.

Reply:
Love it